xarray/plot/dataset_plot.py

```diff
@@ -721,8 +721,8 @@ def _temp_dataarray(ds: Dataset, y: Hashable, locals_: dict[str, Any]) -> DataAr
     """Create a temporary datarray with extra coords."""
     from xarray.core.dataarray import DataArray
 
-    # Base coords:
-    coords = dict(ds.coords)
+    coords = dict(ds[y].coords)
+    dims = dict.fromkeys(ds[y].dims)
```

Comment:
If you don't use the values anyway, why not use a set instead?

Reply:
Done. Will look even nicer when `ds.dims` returns a set.